src/cargo/core/interning.rs

```diff
@@ -0,0 +1,34 @@
+use std::fmt;
+use std::sync::RwLock;
+use std::collections::HashMap;
+
+lazy_static! {
+    static ref STRING_CASHE: RwLock<(Vec<String>, HashMap<String, usize>)> =
+        RwLock::new((Vec::new(), HashMap::new()));
+}
+
+#[derive(Ord, PartialOrd, Eq, PartialEq, Hash, Clone, Copy)]
```

Comment:
Could the `Ord` and `PartialOrd` references be left out here? I think b/c they're comparing strings rather than contents it may end up being confusing down the road.